spec/models/git_repository_spec.rb

```diff
@@ -345,5 +345,35 @@
         end
       end
     end
+
+    context "check_connection?" do
+      require 'net/ping/external'
+      let(:ext_ping) { instance_double(Net::Ping::External) }
+
+      before do
+        allow(Net::Ping::External).to receive(:new).and_return(ext_ping)
+        allow(ext_ping).to receive(:exception)
+      end
+
+      it "returns true if it can ping the repo" do
+        allow(ext_ping).to receive(:ping?).and_return(true)
+        expect($log).to receive(:debug).with(/pinging '.*' to verify network connection/)
+        expect(repo.check_connection?).to eq(true)
+      end
+
+      it "returns false if it cannot ping the repo" do
+        allow(ext_ping).to receive(:ping?).and_return(false)
+        expect($log).to receive(:debug).with(/pinging '.*' to verify network connection/)
+        expect($log).to receive(:debug).with(/ping failed: .*/)
+        expect(repo.check_connection?).to eq(false)
+      end
+
+      it "handles git urls without issue" do
+        allow(repo).to receive(:url).and_return("git@example.com:ManageIQ/manageiq.git")
```

Comment:
I think we need specs for all possible URLs we can accept (such as `ssh://git@example.com/ManageIQ/manageiq.git`, and `https://example.com/ManageIQ/manageiq.git`)

Reply:
@NickLaMuro Do you have a relatively comprehensive list of the possible URL formats?

Reply:
This is the best I have found for a comprehensive list:

https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats

Which isn't terribly credible, but it is at least in a format that I think is a bit easier to digest, compared to looking at some of the tests from `git`'s source:

https://github.com/git/git/blob/master/t/t0110-urlmatch-normalization.sh

Reply:
I only added a spec for `git@` url's because `URI.parse` cannot deal with it. Any url that it can handle, including `ssh://` url's, doesn't need specs IMO.

Reply:
I think we should add the specs only because someone could come along in the future and change this method and not realize they are breaking something.  It also verifies that the existing code actually works with those other formats (I actually though of it because you replaced `git@` with `https://` (without seeing the `starts_with?` bit), and though it was going to convert `ssh://git@example.com` to `ssh://https://example.com`

Reply:
However, @djberg96, I agree maybe we don't need _all_...I'd be happy with an `ssh://`, `ssh://git@`, `https://` and probably `file://`

Reply:
@Fryguy ok, updated.